micro_sam/sam_3d_wrapper.py

```diff
@@ -0,0 +1,151 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from typing import Type
+#from segment_anything import build_sam, SamPredictor
+#from segment_anything import sam_model_registry
+from segment_anything.modeling.image_encoder import window_partition, window_unpartition
+from segment_anything.modeling.image_encoder import Block as SamBlock
+from segment_anything.modeling import Sam
+from segment_anything import SamPredictor
+
+
+class Predictor3D(SamPredictor):
+    def __init__(
+        self,
+        #predictor: SamPredictor,
+        sam_model: Sam,
+        d_size,
+        model_type: str = "vit_b",
+    ):
+        super().__init__(sam_model)
+        self.model_type = model_type
+        self.d_size = d_size
+
+        # predictor.model = Sam3DWrapper(predictor.model, self.d_size)
+        # self.predictor = predictor
+
+
+class Sam3DWrapper(nn.Module):
+    def __init__(
+        self,
+        sam_model: Sam,
+        d_size,
+    ):
+        super().__init__()
+        self.sam_model = sam_model
+        self.d_size = d_size
```

Comment:
What exactly is `d_size`? Can you add a doc-string here that explains it?

Reply:
added docstring